lib/src/view/game/game_player.dart

```diff
@@ -61,10 +61,14 @@ class GamePlayer extends StatelessWidget {
   final Duration? timeToMove;
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final remaingHeight = estimateRemainingHeightLeftBoard(context);
     final playerFontSize = remaingHeight <= kSmallRemainingHeightLeftBoardThreshold ? 14.0 : 16.0;
 
+    final player = side == Side.white ? game.white : game.black;
+
+    final session = ref.watch(authSessionProvider);
```

Comment:
Instead of watching the session here, I think we should use the `youAre`, `me` and `opponent` props of game.

They are already part of the playable and archived game interface, but the code is duplicated and not in the `BaseGame` interface. It only needs a slight refactoring to fix that.

Reply:
Done! While I was at it, I also moved `finished`, `aborted` and `playable` fields to `BaseGame`, as these were also duplicated in classes using the `BaseGame` mixin